### PR TITLE
fix notarizing code 

### PR DESCRIPTION
**Pull request by nammn**
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### Summary
fixes notarzing code 
- use correct go version
- fix wrongly used paths for notarizing for goreleaser 

evg: https://spruce.mongodb.com/task/mongodb_enterprise_kubernetes_release_mcli_package_goreleaser_patch_b0a80faa1870a0b6432d63897f9397130d933768_65e7ffde2fbabe21948a9640_24_03_06_05_32_14/logs?execution=0

downloading https://github.com/mongodb/mongodb-enterprise-kubernetes/releases/tag/untagged-935fdcdf7b8d6f2acf91 shows that the binary works without mac complaining.

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -42,24 +45,26 @@ functions:
...
           set -Eeu pipefail
-          
-          ${workdir}/goreleaser release
+
+          export PATH=$GOROOT/bin:$PATH
+          ${workdir}/goreleaser release --rm-dist
 
 tasks:
   - name: package_goreleaser
+    git_tag_only: true
```
Comment (nammn) [inline .evergreen.yml:67]: according to our evergreen setting page and this documentation: https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-and-Distro-Settings/#triggering-versions-with-git-tags this should suffice. 

`mms-build-account` is part of the allowed users to create tags. So we will have the following release process:
- mms-build-account creates a new release tag
- evergreen picks that up and will run this tasks/variants
- profit? 💸

File: .evergreen.yml [hunk 2]
```diff
@@ -42,24 +45,27 @@ functions:
...
         working_dir: src/github.com/mongodb/mongodb-enterprise-kubernetes/tools/multicluster
-        add_to_path:
-          - src/github.com/mongodb/tools
-          - src/github.com/mongodb/tools/linux_amd64
         include_expansions_in_env:
           - GITHUB_TOKEN
           - macos_notary_keyid
           - macos_notary_secret
           - workdir
+          - triggered_by_git_tag
```
Comment (nammn) [inline .evergreen.yml:53]: > ${triggered_by_git_tag} is the name of the tag that triggered this version, if applicable

File: .evergreen.yml [hunk 3]
```diff
@@ -42,24 +45,27 @@ functions:
...
         env:
+          <<: *go_env
           MACOS_NOTARY_KEY: ${macos_notary_keyid}
           MACOS_NOTARY_SECRET: ${macos_notary_secret}
+          GORELEASER_CURRENT_TAG: ${triggered_by_git_tag}
```
Comment (nammn) [inline .evergreen.yml:58]: just copy paste in what we had prior

**Review by lsierant — APPROVED** [review]
Awesome! LGTM!

**Review by mircea-cosbuc — APPROVED** [review]

**Pull request merged**